engine/class_modules/sc_shaman.cpp

```diff
@@ -6415,12 +6417,15 @@ void shaman_t::create_buffs()
   buff.ascendance = new ascendance_buff_t( this );
   buff.ghost_wolf = make_buff( this, "ghost_wolf", find_class_spell( "Ghost Wolf" ) );
 
-  buff.elemental_blast_crit = make_buff<stat_buff_t>( this, "elemental_blast_critical_strike", find_spell( 118522 ) );
-  buff.elemental_blast_crit->set_max_stack( 1 );
-  buff.elemental_blast_haste = make_buff<stat_buff_t>( this, "elemental_blast_haste", find_spell( 173183 ) );
-  buff.elemental_blast_haste->set_max_stack( 1 );
-  buff.elemental_blast_mastery = make_buff<stat_buff_t>( this, "elemental_blast_mastery", find_spell( 173184 ) );
-  buff.elemental_blast_mastery->set_max_stack( 1 );
+  buff.elemental_blast_crit = make_buff( this, "elemental_blast_critical_strike", find_spell( 118522 ) )
+          ->set_max_stack(1)
+          ->add_invalidate(CACHE_CRIT_CHANCE);
+  buff.elemental_blast_haste = make_buff( this, "elemental_blast_haste", find_spell( 173183 ) )
+          ->set_max_stack(1)
+          ->add_invalidate(CACHE_HASTE);
+  buff.elemental_blast_mastery = make_buff( this, "elemental_blast_mastery", find_spell( 173184 ) )
+          ->set_max_stack(1)
+          ->add_invalidate(CACHE_MASTERY);
```

Comment:
Unlike stat buffs, regular buffs make no attempt to extract the crit/haste/etc from the spell data. You need to set the default values here in some way (set_default_value, set_default_value_from_effect, set_default_value_from_effect_type) so that you can use check_value in the later code.